components/places/src/error.rs

```diff
@@ -220,6 +220,15 @@ pub enum PlacesError {
     /// - Attempting to delete any of the bookmark roots.
     #[error("CannotUpdateRoot error: {0}")]
     CannotUpdateRoot(String),
+
+    // XX - We should kill this and replace
+    // it with other specific errors
```

Comment:
I'm not sure if your comment implies the desire for a group of observation specific errors or a unexpected specific errors.

If you'd like to make this a specific error, that can work as part of this patch now as well. Using `UnexpectedPlacesException` is the only special exception as you can see here: https://github.com/mozilla-mobile/android-components/blob/0559e5a47e708b2f450d50621590c3e605486012/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorage.kt#L85-L94=

Reply:
I think Tarik is saying that having `InternalError` is a smell and ideally would not exist, because all the errors which fall into this category should have their own errors - but that's probably to intrusive to be actionable TBH (and chatting to Tarik, we are likely to go the other way - kill all those other internal errors, which will have the same effect)

Regardless, I agree this comment should be clarified.

(I don't think we've identified a new error specific to history metadata though)